tests/cases/fourslash/server/convertFunctionToEs6Class-server2.ts

```diff
@@ -16,16 +16,16 @@
 verify.codeFix({
     description: "Convert function to an ES2015 class",
     newFileContent:
-`/**\r
- * JSDoc Comment\r
- */\r
+`/**
+ * JSDoc Comment
+ */

```

Comment:
This _feels_ like maybe a bug? It's switching the type of newlines during the refactor

Reply:
@orta Thanks for the feedback 👍. It uses default formatting because we replacing node in-place and avoid copying comments block and creating a new node. What do you think does that make sense? Or do we need to copy comments?

Reply:
Yeah, let's copy the comment - that should be enough IMO 👍

Reply:
Sorry, I didn't add clarification about my question :).

In the current implementation, there are three steps to insert a new class declaration
1. copy comments., I think in this stage `copyLeadingComments` changes comment and adds `\r` 
2. delete node (with comments, for that reason need to copy comments before)
3. insert a new node

As I understood from the https://github.com/microsoft/TypeScript/issues/23871 description, better to combine three steps into one to just replace a node with a function by class declaration. Maybe I'm wrong and need to do the following steps 

1. copy comment before (to change comment with a carriage return) and remove (to avoid comment duplication) previous original comment (I'm not sure, however, we can try to use `{ leadingTriviaOption: textChanges.LeadingTriviaOption.IncludeAll }` in `replaceNode`)
2. replace a node with function by a class declaration

Do `fourslash/server` tests require explicit changing comments to add carriage return symbols?

Reply:
@orta Sorry, I didn't add clarification about my question :).

In the current implementation, there are three steps to insert a new class declaration
1. copy comments., I think in this stage `copyLeadingComments` changes comment and adds `\r` 
2. delete node (with comments, for that reason need to copy comments before)
3. insert a new node

As I understood from the https://github.com/microsoft/TypeScript/issues/23871 description, better to combine three steps into one to just replace a node with a function by class declaration. Maybe I'm wrong and need to do the following steps 

1. copy comment before (to change comment with a carriage return) and remove (to avoid comment duplication) previous original comment (I'm not sure, however, we can try to use `{ leadingTriviaOption: textChanges.LeadingTriviaOption.IncludeAll }` in `replaceNode`)
2. replace a node with function by a class declaration

Do `fourslash/server` tests require explicit changing comments to add carriage return symbols?

Reply:
@orta could you check my question?

Reply:
@orta Could you check my previous comment?

Reply:
Argh, I didn't notice your comment ( I originally thought I just wrote a really long one because your avatar is white 💯  )

No, I think this is acceptable based on the original spec. Let's get it in.